ClientFactory/ClientFactoryInterface.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Http\HttplugBundle\ClientFactory;
+
+use Http\Client\HttpClient;
+
+/**
+ * Interface ClientFactoryInterface.
+ */
+interface ClientFactoryInterface
+{
+    /**
+     * Configure and return a client.
+     *
+     * @param array $config
+     *
+     * @return HttpClient
```

Comment:
Then InvalidConfiguration or something like that, but ClientWasNotCreated does not tell too much about the "WHY" in my opinion.